test/CMakeLists.txt

```diff
@@ -8,8 +8,8 @@ macro(build_tests TESTS)
   foreach(TEST ${TESTS})
     get_filename_component(TARGET_NAME ${TEST} NAME_WE)
     add_executable(${TARGET_NAME} ${TEST})
-    set_target_properties(${TARGET_NAME} PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
-    target_link_libraries(${TARGET_NAME} AdePT_cuda AdePT_G4_integration AdePT)
+    # NB: We do not use PRIVATE for executables due to an apparent limitation in CudaRDCUtils
```

Comment:
The output from the `cmake` step if adding `PRIVATE` here:

```console
CMake Error at cmake/CudaRdcUtils.cmake:850 (target_link_libraries):
  The keyword signature for target_link_libraries has already been used with
  the target "test_atomic".  All uses of target_link_libraries with a target
  must be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * cmake/CudaRdcUtils.cmake:738 (target_link_libraries)

Call Stack (most recent call first):
  test/CMakeLists.txt:12 (cuda_rdc_target_link_libraries)
  test/CMakeLists.txt:76 (build_tests)
```

The `target_link_libraries` at 850 is "bare" and could probably use a direct `PRIVATE` though I do recall there are use cases for `PUBLIC` linking of executables (plugins?) if only for CPU.

Reply:
Is there an actual difference on the link line of the executable between the 2 settings?  (I naively thought it would only be a setting for dependency propagation in `CMake` and thus 'useless' for executable).